File: Assets/Zombie/Model/ZombieAI.js
#pragma strict

var Distance :float ;
var Target : Transform; 
var lookAtDistance :float = 20;
var chaseRange :float = 10;
var attackRange :float = 2.2;
var moveSpeed :float = 3;
var Damping :float =6;
var attackRepeatTime :float = 1;

var TheDammage :float = 10;

private var attackTime : float;

var controller : CharacterController;
var gravity : float = 20;

private var moveDirection : Vector3 = Vector3.zero;

function Start () {
    attackTime = Time.time;
    Findhealth ();
}

function Update () {
    Distance = Vector3.Distance(Target.position , transform.position);

    if (Distance< lookAtDistance){
        lookAt();
    }
    if (Distance < attackRange){
        attack();
    }
    else if(Distance < chaseRange) {
        chase();
    }

}
function lookAt(){
    var rotation = Quaternion.LookRotation(Target.position - transform.position );
    transform.rotation = Quaternion.Slerp(transform.rotation, rotation,Time.deltaTime * Damping);
        
}

function chase (){
    GetComponent.<Animator>().Play("walk");

    moveDirection = transform.forward;
    moveDirection *= moveSpeed;

    moveDirection.y -= gravity * Time.deltaTime;
    controller.Move(moveDirection*Time.deltaTime);

}
function attack (){
    if(Time.time > attackTime){
        GetComponent.<Animator>().Play("attack");

       	Target.SendMessage("ApplyDammage", TheDammage);
        Debug.Log("The enemy has attacked ");
        attackTime = Time.time + attackRepeatTime;
    }

}

function ApplyDammage(){
    chaseRange += 30;
    moveSpeed+= 2;
    lookAtDistance +=40;
}

function Findhealth()
{
    Target = GameObject.FindGameObjectWithTag ("Player").transform;
}

File: Assets/map2/ZombieAI.cs
using UnityEngine;
using System.Collections;

public class ZombieAI : MonoBehaviour {

	public float Distance;
	public Transform Target; 
	public float lookAtDistance = 20f;
	public float chaseRange = 10f;
	public float attackRange = 2.2f;
	public float moveSpeed = 3f;
	public float Damping = 6f;
	public float attackRepeatTime = 1f;

	public float TheDammage = 10f;

	private float attackTime;

	public CharacterController controller;
	public float gravity = 20f;

	private Vector3 moveDirection = Vector3.zero;
	// Use this for initialization
	void Start () {
		attackTime = Time.time;
		Findhealth ();
	}
	
	// Update is called once per frame
	void Update () {
		Distance = Vector3.Distance(Target.position , transform.position);

		if (Distance < lookAtDistance) {
			lookAt();
		}
		if (Distance < attackRange){
			attack();
		}
		else if(Distance < chaseRange) {
			chase();
		}	
	}

	void lookAt(){
		Quaternion rotation = Quaternion.LookRotation(Target.position - transform.position);
		transform.rotation = Quaternion.Slerp(transform.rotation, rotation, Time.deltaTime * Damping);

	}

	void chase (){
		GetComponent<Animator>().Play("walk");

		moveDirection = transform.forward;
		moveDirection *= moveSpeed;

		moveDirection.y -= gravity * Time.deltaTime;
		controller.Move(moveDirection*Time.deltaTime);

	}
	void attack (){
		if(Time.time > attackTime){
			GetComponent<Animator>().Play("attack");

			Debug.Log("The enemy has attacked ");
			attackTime = Time.time + attackRepeatTime;
		}

	}

	void ApplyDammage(){
		chaseRange += 30f;
		moveSpeed+= 2f;
		lookAtDistance += 40f;
	}

	void Findhealth()
	{
		Target = GameObject.FindGameObjectWithTag("Player").transform;
	}
}
